src/pilot_simulation.jl

```diff
@@ -1,21 +1,145 @@
+V = sose.get_profile_time_series(ds3, "VVEL",  lat, lon, days) |> Array{FT}
+Θ = sose.get_profile_time_series(ds3, "THETA", lat, lon, days) |> Array{FT}
+S = sose.get_profile_time_series(ds3, "SALT",  lat, lon, days) |> Array{FT}
+
+ds2.close()
+ds3.close()
+
+#####
+##### Create linear interpolations for base state solution
+#####
+
+ts = day * (0:days-1) |> collect
+zC = ds3.Z.values
+
+ℑτx = interpolate((ts,), τx, Gridded(Linear()))
+ℑτy = interpolate((ts,), τy, Gridded(Linear()))
+ℑQθ = interpolate((ts,), Qθ, Gridded(Linear()))
+ℑQs = interpolate((ts,), Qs, Gridded(Linear()))
+
+# z coordinate needs to be in increasing order.
+reverse!(zC)
+U = reverse(U, dims=2)
+V = reverse(V, dims=2)
+Θ = reverse(Θ, dims=2)
+S = reverse(S, dims=2)
+
+ℑU = interpolate((ts, zC), U, Gridded(Linear()))
+ℑV = interpolate((ts, zC), V, Gridded(Linear()))
+ℑΘ = interpolate((ts, zC), Θ, Gridded(Linear()))
+ℑS = interpolate((ts, zC), S, Gridded(Linear()))
+
+#####
+##### Set up the grid
+#####
+
+Nx = Ny = Nz = 32
+Lx = Ly = Lz = 100
+topology = (Periodic, Bounded, Bounded)
+grid = RegularCartesianGrid(topology=topology, size=(Nx, Ny, Nz) x=(0, Lx), y=(0, Ly), z=(-Lz, 0))
+
+#####
+##### Set up forcing forcings to
+#####   1. include mean flow interactions in the momentum equation, and
+#####   2. weakly relax tracer fields to the base state.
+#####
+
+# Fu′ = - w′∂z(U) - U∂x(u′) - V∂y(u′)
+# FIXME? Do we need to use the flux form operator ∇·(Uũ′) instead of ũ′·∇U ?
+@inline Fu′(i, j, k, grid, t, ũ′, c′, p) =
+    @inbounds - ũ′.w[i, j, k] * ∇(p.ℑU, t, grid.zC[k])[2] - p.ℑU(t, grid.zC[k]) * ∂xᶜᵃᵃ(i, j, k, grid, ũ′.u) - p.ℑV(t, grid.zC[k]) * ∂yᵃᶜᵃ(i, j, k, grid, ũ′.u)
+
+# Fv′ = - w′∂z(V) - U∂x(v′) - V∂y(v′)
+@inline Fv′(i, j, k, grid, t, ũ′, c′, p) =
+    @inbounds - ũ′.w[i, j, k] * ∇(p.ℑV, t, grid.zC[k])[2] - p.ℑU(t, grid.zC[k]) * ∂xᶜᵃᵃ(i, j, k, grid, ũ′.v) - p.ℑV(t, grid.zC[k]) * ∂yᵃᶜᵃ(i, j, k, grid, ũ′.v)
+
+# Fw′ = - U∂x(w′) - V∂y(w′)
+@inline Fw′(i, j, k, grid, t, ũ′, c′, p) =
+    @inbounds - p.ℑU(t, grid.zC[k]) * ∂xᶜᵃᵃ(i, j, k, grid, ũ′.u) - p.ℑV(t, grid.zC[k]) * ∂yᵃᶜᵃ(i, j, k, grid, ũ′.v)
+
+# Fθ′ = - ∂t(Θ) - U∂x(θ′) - V∂y(θ′) - w′∂z(Θ)
+# FIXME? Do we need to include the ∂t(Θ) terms?
+@inline Fθ′(i, j, k, grid, t, ũ′, c′, p) =
+    @inbounds - p.ℑU(t, grid.zC[k]) * ∂xᶠᵃᵃ(i, j, k, grid, c′.T) - p.ℑV(t, grid.zC[k]) * ∂yᵃᶜᵃ(i, j, k, grid, ũ′.v) - ũ′.w[i, j, k] * ∇(p.ℑΘ, t, grid.zC[k])[2]
+
+# Fs′ = - ∂t(S) - U∂x(s′) - V∂y(s′) - w′∂z(S)
+@inline Fs′(i, j, k, grid, t, ũ′, c′, p) =
+    @inbounds - p.ℑU(t, grid.zC[k]) * ∂xᶠᵃᵃ(i, j, k, grid, c′.S) - p.ℑV(t, grid.zC[k]) * ∂yᵃᶜᵃ(i, j, k, grid, ũ′.v) - ũ′.w[i, j, k] * ∇(p.ℑS, t, grid.zC[k])[2]
+
+# Timescale for relaxation to large-scale solution.
+week = 7day
+μ = (T=1/week, S=1/week)
+
+# FIXME: Should be μ(C - c̅) so I need to add horizontal averages to parameters.
+@inline Fθ_μ(i, j, k, grid, t, ũ′, c′, p) = @inbounds p.μ.T * (p.ℑΘ(t, grid.zC[k]) - c′.T[i, j, k])
```

Comment:
the distinction may not matter for weak relaxation: the purpose of using the horizontal average is to avoid damping turbulent fluctuations